### PR TITLE
chore: Discovery cluster test label

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Needed a new label to identify testing resource with their test clusters.

**How was this change tested?**
- N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/test/metadata.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ import (
 )
 
 const DiscoveryLabel = v1alpha5.TestingGroup + "/test-id"
+const ClusterLabel = v1alpha5.TestingGroup + "/cluster"
```
Comment (ellistarn) [inline pkg/test/metadata.go:32]: Possible to collapse these two concepts?

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

File: pkg/test/metadata.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ import (
 	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
 )
 
-const DiscoveryLabel = v1alpha5.TestingGroup + "/test-id"
+const DiscoveryLabel = v1alpha5.TestingGroup + "/cluster-id"
```
Comment (jonathan-innis) [inline pkg/test/metadata.go:31]:
```suggestion
const DiscoveryLabel = v1alpha5.TestingGroup + "/cluster"
```

Why not just `cluster`?

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**